expression/builtin_string.go

```diff
@@ -1979,25 +1979,38 @@ type toBase64FunctionClass struct {
 }
 
 func (c *toBase64FunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinToBase64Sig{newBaseBuiltinFunc(args, ctx)}
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpString, tpString)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	bf.tp.Flen = base64NeededEncodedLength(bf.args[0].GetType().Flen)
+	sig := &builtinToBase64Sig{baseStringBuiltinFunc{bf}}
 	return sig.setSelf(sig), errors.Trace(c.verifyArgs(args))
 }
 
 type builtinToBase64Sig struct {
-	baseBuiltinFunc
+	baseStringBuiltinFunc
+}
+
+// base64NeededEncodedLength return the base64 encoded string length
+func base64NeededEncodedLength(n int) int {
```

Comment:
@stdupp 
Noop, not the type of arg and return result
what do you mean by NUL termination of string?

MySQL:
``` c++
uint64
base_needed_decoded_length(uint64 length_of_encoded_data) {
  return (uint64) ceil(length_of_encoded_data * 3 / 4)
}
```
what does `length + (length - 1)/76` means?